src/passes/Print.cpp

```diff
@@ -119,59 +131,104 @@ std::ostream& operator<<(std::ostream& os, SigName sigName) {
             }
             os << "]";
             continue;
+          } else if (heapType.isStruct()) {
+            auto struct_ = heapType.getStruct();
+            os << "{";
+            auto sep = "";
+            for (auto& field : struct_.fields) {
+              os << sep;
+              sep = "_";
+              if (field.mutable_) {
+                os << "mut:";
+              }
+              printType(field.type);
+            }
+            os << "}";
+            continue;
+          } else if (heapType.isArray()) {
+            os << "[";
+            auto element = heapType.getArray().element;
+            if (element.mutable_) {
+              os << "mut:";
+            }
+            printType(element.type);
+            os << "]";
+            continue;
```

Comment:
Do we support packed types here? We seem to support it in wasm-type.cpp, but in type name printing here I mean.

Reply:
It may be missing here, yes. In a later PR I'll be adding it, when we add an instruction that can also use them (and has `_s/_u` variants).